Modules/Core/Common/include/itkVector.h

```diff
@@ -129,6 +129,10 @@ class ITK_TEMPLATE_EXPORT Vector : public FixedArray<T, NVectorDimension>
   Vector(const Vector<TVectorValueType, NVectorDimension> & r)
     : BaseArray(r)
   {}
+  template <typename TVectorValueType>
+  Vector(const FixedArray<TVectorValueType, NVectorDimension> & r)
+    : BaseArray(r)
```

Comment:
If possible, I would declare this constructor `explicit`. Because I think the original idea was that every Vector is a FixedArray, but _not_ every FixedArray is a Vector.

Reply:
Not sure where the error lies: do we test somewhere that the `-` operator can be applied to the basic underlying `itk::Point` types with different pixel types? As simple and trivial as it may seem when `float` and `double` are involved, maybe doing that in an `itkPointTest` may help in discarding that as the cause.

Reply:
Or maybe it is closer to what Niels mentions, and the problem does not lie in the subtraction, but the assignation of the result to an `itk::VectorType`. Maybe it has to be done in a `for` loop to avoid the warning?